storage/storage-interface/src/finality_view.rs

```diff
@@ -27,6 +27,15 @@ impl FinalityView {
 }
 
 impl FinalityView {
+    /// Returns the height of the previously finalized block.
+    ///
+    /// If [`set_finalized_block_height`] has never been called on this
+    /// instance, `None` is returned.
+    pub fn finalized_block_height(&self) -> Option<u64> {
+        let fin_ledger_info = self.finalized_ledger_info.read().unwrap();
```

Comment:
are we sure we want to `unwrap` ?

Reply:
That's a lock acquisition. If it returns an error, the lock is poisoned, meaning some other thread has crashed while holding the lock. So it's a condition fatal enough to crash in turn.

Reply:
Technically, it's an invariant violation for us not to do this. But, I think maybe we should log and then panic to be a little less opaque.